pyquil/api/qvm.py

```diff
@@ -193,51 +211,63 @@ def wavefunction(self, quil_program, classical_addresses=None):
 
         :param Program quil_program: A Quil program.
         :param list|range classical_addresses: An optional list of classical addresses.
+        :param needs_compilation: If True, preprocesses the job with the compiler.
+        :param isa: If set, compiles to this target ISA.
         :return: A tuple whose first element is a Wavefunction object,
                  and whose second element is the list of classical bits corresponding
                  to the classical addresses.
         :rtype: Wavefunction
         """
+        if needs_compilation:
+            raise TypeError("Synchronous calls do not support compilation preprocessing. Use wavefunction_async instead.")
```

Comment:
Same thing here with meeting the user halfway

Reply:
Sounds good. I put in a branch to switch over to the async endpoint, and I have it emit a warning that this maybe isn't what the user expects.